lib/services/parser.go

```diff
@@ -518,3 +522,139 @@ func newParserForIdentifierSubcondition(ctx RuleContext, identifier string) (pre
 		},
 	})
 }
+
+// NewResourceParser returns a parser made for boolean expressions based on a
+// json-serialiable resource. Customized to allow short identifiers common in all
+// resources:
+//  - `metadata.name` can be referenced with `name` ie: `name == "jenkins"``
+//  - `metadata.labels + spec.dynamic_labels` can be referenced with `labels`
+//     ie: `labels.env == "prod"`
+// All other fields can be referenced by starting expression with identifier `resource`
+// followed by the names of the json fields ie: `resource.spec.public_addr`.
+func NewResourceParser(resource types.ResourceWithLabels) (BoolPredicateParser, error) {
+	eqPred := func(a interface{}, b interface{}) predicate.BoolPredicate {
+		return func() bool {
+			return isOperandsEqual(a, b)
+		}
+	}
+	neqPred := func(a interface{}, b interface{}) predicate.BoolPredicate {
+		return func() bool {
+			return !isOperandsEqual(a, b)
+		}
+	}
+
+	p, err := predicate.NewParser(predicate.Def{
+		Operators: predicate.Operators{
+			AND: predicate.And,
+			OR:  predicate.Or,
+			NOT: predicate.Not,
+			EQ:  eqPred,
```

Comment:
```suggestion
			EQ:  isOperandsEqual,
```

Reply:
`eqPred` is necessary as it returns a closure which is expected by the parser